### ISSUE TITLE
Stacked Denoising Autoencoders pretraining input IN() error

### ISSUE THREAD

**Issue by amit4111989** [lisa-lab/DeepLearningTutorials#138]
The inputs (corruption and learning rate) are an instance of In() which appears to be a part of old semantics where updates are not available.
I changed it to to Param() and it worked for me. 
Please make the required changes

**Comment by nouiz** [comment]
We removed Param. In now support all what Param was doing.

Maybe you need to update Theano? It work with the dev version of Theano. What did you had?

**Comment by lamblin** [comment]
That is indeed the other way around. Your version of Theano is too old for the tutorials, so we made the changes for the compatibility with the latest development version of Theano.
